compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -1994,8 +1995,24 @@ public BLangNode transform(FunctionCallExpressionNode functionCallNode) {
 
     @Override
     public BLangNode transform(ErrorConstructorExpressionNode errorConstructorExprNode) {
-        return createBLangInvocation(errorConstructorExprNode.errorKeyword(), errorConstructorExprNode.arguments(),
-                getPosition(errorConstructorExprNode), false);
+        BLangErrorConstructorExpr errorConstructorExpr =
+                (BLangErrorConstructorExpr) TreeBuilder.createErrorConstructorExpressionNode();
+        errorConstructorExpr.pos = getPosition(errorConstructorExprNode);
+        if (errorConstructorExprNode.typeReference().isPresent()) {
+            errorConstructorExpr.errorTypeRef =
+                    (BLangUserDefinedType) createTypeNode(errorConstructorExprNode.typeReference().get());
+        }
+
+        List<BLangExpression> positionalArgs = errorConstructorExpr.positionalArgs;
+        List<BLangNamedArgsExpression> namedArgs = errorConstructorExpr.namedArgs;
```

Comment:
Wouldn't it be clearer if we define two new lists here, populate them in the for loop, and then set

```java
        errorConstructorExpr.positionalArgs = positionalArgs;
        errorConstructorExpr.namedArgs = namedArgs;
```

before returning `errorConstructorExpr`?

Reply:
Fixed